stix2/datastore/taxii.py

```diff
@@ -235,9 +272,12 @@ def query(self, query=None, version=None, _composite_filters=None):
             query.remove(taxii_filters)
             all_data = list(apply_common_filters(all_data, query))
 
-        except HTTPError:
+        except HTTPError as e:
             # if resources not found or access is denied from TAXII server, return empty list
-            all_data = []
+            if e.response.status_code == 404:
```

Comment:
Is there any reason why this `except` handling is different from the one on line 182?

Reply:
I think the idea was that the "singular" getters could return None if the object was not found/status 404.  The "plural" getters raise an error on 404 to disambiguate collection-not-found from collection-was-empty/everything-filtered-out (former uses exception, latter uses empty list).  When looking for a single object, there is no "empty" condition to handle, so it's simpler.  That's my take on it anyway.

Reply:
Read the issue again. Thanks for clarifying.